src/parser/vast_parser.js

```diff
@@ -576,4 +580,32 @@ export class VASTParser extends EventEmitter {
       }
     }
   }
+
+  /**
+   * Calculate the estimated bitrate and add it to the bitrates list
+   * @param {Number} byteLength - The length of the response in bytes.
+   * @param {Number} duration - The duration of the request in ms.
+   */
+  updateEstimatedBitrate(byteLength, duration) {
+    if (!byteLength || !duration || byteLength <= 0 || duration <= 0) {
+      return;
+    }
+
+    // We want the bitrate in bit by seconds, byteLength are in bytes and duration in ms, we need to convert them
+    const bitrate = (byteLength * 8) / duration * 1000;
+    this.estimatedBitrates.push(bitrate);
+  }
+
+  /**
+   * Returns the estimated bitrate calculated from all previous requests
+   * @returns The average of all estimated bitrates.
+   */
+  getEstimatedBitrate() {
+    if (!this.estimatedBitrates.length) {
+      return 0;
+    }
+
+    const sum = this.estimatedBitrates.reduce((acc, val) => acc + val, 0);
```

Comment:
Why not do this calculation while adding a new byterate value? and keep the pondaration (total of byterates)?
This way if the estimation is requested several time we could save some calculation? no more array to loop.
WDYT?

Reply:
done